### DISCUSSION TITLE
Question about checksum

### DISCUSSION THREAD

**Discussion by shyo21** [ANLAB-KAIST/KENSv3#241] [Project Q&A 3. Internet Layer (IP/Routing)] [answered]
![image](https://github.com/ANLAB-KAIST/KENSv3/assets/79900341/4ca6ce6d-92ab-4cb5-b597-48e8f8b6aa11)

Our pa4 document says that we don't need to calculate checksum.

![image](https://github.com/ANLAB-KAIST/KENSv3/assets/79900341/1027114c-5e94-41dd-b645-61be246226b6)

However, in pwospf.txt file, it says that for every incoming packet, checksum must be verified.

I'm a bit confused here :(

Is this mean that we don't need to fill in checksum field when we send packet AND we also don't need to verify checksum when receiving it?

**Comment by Pusnow** [comment, marked as answer]
> Is this mean that we don't need to fill in checksum field when we send packet AND we also don't need to verify checksum when receiving it?

Yes, you don't need to calculate/verify checksum on sending/receiving packets.
